common/resource/resourceImpl.go

```diff
@@ -568,38 +573,23 @@ func (h *Impl) GetPartitioner() partition.Partitioner {
 	return h.partitioner
 }
 
-func mapIGs(log log.Logger, in []interface{}) []string {
-	var allIsolationGroups []string
-	for k := range in {
-		v, ok := in[k].(string)
-		if ok {
-			allIsolationGroups = append(allIsolationGroups, v)
-		} else {
-			log.Error("Invalid isolation-group: ", tag.Dynamic("isolation-group", v))
-		}
-	}
-	return allIsolationGroups
-}
-
 // Use the provided IsolationGroupStateHandler or the default one
-func ensureIsolationGroupStateHandlerOrDefault(params *Params,
-	cfgStoreMgr persistence.ConfigStoreManager,
+func ensureIsolationGroupStateHandlerOrDefault(
+	params *Params,
 	dc *dynamicconfig.Collection,
 	domainCache cache.DomainCache,
 ) isolationgroup.State {
 
-	allIGs := dc.GetListProperty(dynamicconfig.AllIsolationGroups)()
-	allIsolationGroups := mapIGs(params.Logger, allIGs)
-
 	if params.IsolationGroupState != nil {
 		return params.IsolationGroupState
 	}
-	cfg := isolationgroup.Config{
-		IsolationGroupEnabled: dc.GetBoolPropertyFilteredByDomain(dynamicconfig.EnableTasklistIsolation),
-		UpdateFrequency:       dc.GetDurationProperty(dynamicconfig.IsolationGroupStateRefreshInterval),
-		AllIsolationGroups:    allIsolationGroups,
+
+	ig, err := isolationgroup.NewDefaultIsolationGroupStateWatcher(params.Logger, dc, &params.PersistenceConfig, domainCache)
+	if err != nil {
```

Comment:
I think we should return the error and fail the start of the service.

Reply:
There will be scenarios for which it won't be available (SQL DBs, for example). I'm not sure we want to do make this a requirement. 

I understand that you might be relying on it being available in the matching changes. Open to discussing, but I'd suggest we want to take the availability of the library as part of the decision to enable the feature.

Reply:
if it can be `nil`, should `partitioner` be treated in the same way? Because it depends on it